src/main/java/uk/gov/gchq/palisade/resource/AbstractResource.java

```diff
@@ -19,32 +19,38 @@
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 
+import uk.gov.gchq.palisade.Generated;
 import uk.gov.gchq.palisade.ToStringBuilder;
 
 import java.util.Comparator;
-import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
 
 public abstract class AbstractResource implements Resource {
+
+    private static Comparator<Resource> comp = Comparator.comparing(Resource::getId);
     protected String id;
 
     public AbstractResource() {
     }
 
     public AbstractResource id(final String id) {
-        Objects.requireNonNull(id, "The ID of a resource cannot be set to null.");
+        requireNonNull(id, "The ID of a resource cannot be set to null.");
         this.id = id;
```

Comment:
use setters

Reply:
Ok changed the reference in the constructor and the equals method.